integration/test_project_management.py

```diff
@@ -34,22 +44,144 @@ def test_android_app(default_app):
         package_name=TEST_APP_PACKAGE_NAME, display_name=TEST_APP_DISPLAY_NAME_PREFIX)
 
 
-def test_get_android_app_metadata(test_android_app, project_id):
+@pytest.fixture(scope='module')
+def test_ios_app(default_app):
+    del default_app
+    ios_apps = project_management.list_ios_apps()
+    for ios_app in ios_apps:
+        if ios_app.get_metadata().display_name.startswith(TEST_APP_DISPLAY_NAME_PREFIX):
+            return ios_app
+    return project_management.create_ios_app(
+        bundle_id=TEST_APP_BUNDLE_ID, display_name=TEST_APP_DISPLAY_NAME_PREFIX)
+
+
+def test_create_android_app_already_exists(test_android_app):
+    del test_android_app
+
+    with pytest.raises(project_management.ApiCallError) as excinfo:
+        project_management.create_android_app(
+            package_name=TEST_APP_PACKAGE_NAME, display_name=TEST_APP_DISPLAY_NAME_PREFIX)
+    assert 'The resource already exists' in str(excinfo.value)
+    assert excinfo.value.detail is not None
+
+
+def test_android_set_display_name_and_get_metadata(test_android_app, project_id):
     app_id = test_android_app.app_id
+    android_app = project_management.android_app(app_id)
+    new_display_name = '{0} helloworld {1}'.format(
+        TEST_APP_DISPLAY_NAME_PREFIX, random.randint(0, 10000))
+
+    android_app.set_display_name(new_display_name)
     metadata = project_management.android_app(app_id).get_metadata()
+    android_app.set_display_name(TEST_APP_DISPLAY_NAME_PREFIX)  # Revert the display name.
 
     assert metadata.name == 'projects/{0}/androidApps/{1}'.format(project_id, app_id)
     assert metadata.app_id == app_id
     assert metadata.project_id == project_id
-    assert metadata.display_name == TEST_APP_DISPLAY_NAME_PREFIX
+    assert metadata.display_name == new_display_name
     assert metadata.package_name == TEST_APP_PACKAGE_NAME
 
 
 def test_list_android_apps(test_android_app):
     del test_android_app
+
     android_apps = project_management.list_android_apps()
+
     for android_app in android_apps:
         if android_app.get_metadata().display_name.startswith(TEST_APP_DISPLAY_NAME_PREFIX):
             found = True
             break
     assert found
+
+
+def test_get_android_app_config(test_android_app, project_id):
+    config = test_android_app.get_config()
+
+    assert '"mobilesdk_app_id": "{0}"'.format(test_android_app.app_id) in config
```

Comment:
What is the format of the `config` string? Can we do a stronger assertion here than this?

Reply:
Done.

Reply:
Done.

Agreed. The format of the Android config is a JSON, while the iOS config is a .plist; I have added the relevant tests.